flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraph.java

```diff
@@ -732,6 +758,18 @@ else if (current == JobStatus.RESTARTING) {
 		}
 	}
 
+	public void stop() {
+		if(jobType == JobType.STREAMING) {
+			for(ExecutionVertex ev : this.getAllExecutionVertices()) {
+				if(ev.getNumberOfInputs() == 0) { // send signal to sources only
+					ev.stop();
+				}
+			}
+		} else {
+			throw new RuntimeException("STOP is only supported by streaming jobs.");
```

Comment:
Sorry @mjsax, my bad. I overlooked the try catch block in the `JobManager`. Should work perfectly fine :-)